src/shared/components/dropdown_auto_refresh/AutoRefreshDropdown.tsx

```diff
@@ -22,172 +22,142 @@ import {AutoRefresh, AutoRefreshStatus} from 'src/types'
   onChoose: (milliseconds: number) => void
   onManualRefresh?: () => void
   showAutoRefresh?: boolean
+  customRefreshIntervals?: AutoRefreshOption[]
+  customClass?: string
 }
 
-@ErrorHandling
-export default class AutoRefreshDropdown extends Component<Props> {
-  public static defaultProps = {
-    showAutoRefresh: true,
-    selected: {
-      interval: 0,
-      status: 'paused',
-    },
+const AutoRefreshDropdown: FC<Props> = ({
+  selected = {
+    interval: 0,
+    status: 'paused',
+  },
+  onChoose,
+  onManualRefresh,
+  showAutoRefresh = true,
+  customRefreshIntervals = [],
+  customClass,
+}) => {
+  const isPausedOrDisabled =
+    selected.status === AutoRefreshStatus.Paused ||
+    selected.status === AutoRefreshStatus.Disabled
+
+  const setClassName = () => {
+    return classnames('autorefresh-dropdown', {
+      paused: isPausedOrDisabled,
+    })
   }
 
-  constructor(props) {
-    super(props)
+  const refreshIntervals = [
+    ...autoRefreshOptions,
+    ...(customRefreshIntervals ?? []),
+  ]
 
-    this.state = {
-      isOpen: false,
-    }
+  const handleDropdownChange = (autoRefreshOption: AutoRefreshOption): void => {
+    onChoose(autoRefreshOption.milliseconds)
   }
 
-  public render() {
-    if (!this.props.showAutoRefresh) {
-      return <div className={this.className}>{this.manualRefreshButton}</div>
-    }
+  if (!showAutoRefresh) {
     return (
-      <div className={this.className}>
-        <Dropdown
-          style={{width: `${this.dropdownWidthPixels}px`}}
-          button={(active, onClick) => (
-            <Dropdown.Button
-              active={active}
-              onClick={onClick}
-              status={this.dropdownStatus}
-              icon={this.dropdownIcon}
-              testID="autorefresh-dropdown--button"
-            >
-              {this.selectedOptionLabel}
-            </Dropdown.Button>
-          )}
-          menu={onCollapse => (
-            <Dropdown.Menu
-              onCollapse={onCollapse}
-              style={{width: `${DROPDOWN_WIDTH_FULL}px`}}
-            >
-              {autoRefreshOptions.map(option => {
-                if (option.type === AutoRefreshOptionType.Header) {
-                  return (
-                    <Dropdown.Divider
-                      key={option.id}
-                      id={option.id}
-                      text={option.label}
-                    />
-                  )
-                }
-
-                return (
-                  <Dropdown.Item
-                    key={option.id}
-                    id={option.id}
-                    testID={option.id}
-                    value={option}
-                    selected={option.id === this.selectedOptionID}
-                    onClick={this.handleDropdownChange}
-                  >
-                    {option.label}
-                  </Dropdown.Item>
-                )
-              })}
-            </Dropdown.Menu>
-          )}
+      <div className={setClassName()}>
```

Comment:
why not just inject the logic inline here, or declare a constant up above that does the same thing rather than calling a function? Derived state rather than an output of a function